src/main/java/bio/terra/service/filedata/DrsService.java

```diff
@@ -196,7 +208,11 @@ public DRSAccessURL getAccessUrlForObjectId(
 
     DrsId drsId = drsIdService.fromObjectId(objectId);
     Snapshot snapshot = snapshotService.retrieve(UUID.fromString(drsId.getSnapshotId()));
-    GoogleProjectResource projectResource = snapshot.getProjectResource();
+
+    BillingProfileModel billingProfileModel =
+        profileService.getProfileById(snapshot.getProfileId(), authUser);
```

Comment:
I know we talked about this but actually, you already have this information with the snapshot and this saves another round trip to the DB:
```suggestion
        snapshot.getFirstSnapshotSource().getDataset().getDatasetSummary().getDefaultBillingProfile();
```